internal/rpc/starknet/types_test.go

```diff
@@ -0,0 +1,276 @@
+			want: BlockId{},
+			err:  ErrInvalidBlockId,
+		},
+		{
+			name: "unexpected JSON type",
+			data: []byte(`{}`),
+			want: BlockId{},
+			err:  ErrInvalidBlockId,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var got BlockId
+			if err := got.UnmarshalJSON(tt.data); !errors.Is(err, tt.err) {
+				t.Errorf("BlockId.UnmarshalJSON() error = %v, wantErr %v", err, tt.err)
+			}
+			assert.DeepEqual(t, got, tt.want, gocmp.Comparer(func(x, y BlockId) bool {
+				xv := reflect.ValueOf(x)
+				yv := reflect.ValueOf(y)
+				if xv.IsZero() || yv.IsZero() {
+					return xv.IsZero() && yv.IsZero()
+				}
+				if x.idType != y.idType {
+					return false
+				}
+				switch x.idType {
+				case blockIdHash:
+					return x.value.(*felt.Felt).Equal(y.value.(*felt.Felt))
+				case blockIdTag:
+					return x.value.(string) == y.value.(string)
+				case blockIdNumber:
+					return x.value.(int64) == y.value.(int64)
+				default:
+					return false
+				}
+			}))
+		})
+	}
+}
+
+func TestStorageKey_UnmarshalJSON(t *testing.T) {
+	tests := []struct {
+		name string
+		data []byte
+		want StorageKey
+		err  bool
+	}{
+		{
+			name: "empty data",
+			data: []byte{},
+			want: "",
+			err:  true,
+		},
+		{
+			name: "invalid storage key",
+			data: []byte(`"0x0ad328a71faf48c5c3857e99f20a77b18522480956d1cd5bff1ff2df3c8b427b"`),
+			want: "",
+			err:  true,
+		},
+		{
+			name: "valid storage key",
+			data: []byte(`"0x01d328a71faf48c5c3857e99f20a77b18522480956d1cd5bff1ff2df3c8b427b"`),
+			want: "0x01d328a71faf48c5c3857e99f20a77b18522480956d1cd5bff1ff2df3c8b427b",
+			err:  false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var got StorageKey
+			if err := got.UnmarshalJSON(tt.data); (err != nil) != tt.err {
+				t.Errorf("StorageKey.UnmarshalJSON(), unexpected error: %v", err)
+			}
+			assert.DeepEqual(t, got, tt.want)
+		})
+	}
+}
+
+func TestStorageKey_Felt(t *testing.T) {
+	tests := []struct {
+		name       string
+		storageKey StorageKey
+		want       *felt.Felt
+	}{
+		{
+			name:       "test 1",
+			storageKey: "0x07352d792298a2578d6ef20e80bce32473fff67b9b12b1bc431982287190291a",
+			want:       new(felt.Felt).SetHex("0x07352d792298a2578d6ef20e80bce32473fff67b9b12b1bc431982287190291a"),
+		},
+		{
+			name:       "test 2",
+			storageKey: "0x00898cca7dbf84c2213f3a00e84775013bc991bc104d3a00952ce4bc166a4a1a",
+			want:       new(felt.Felt).SetHex("0x00898cca7dbf84c2213f3a00e84775013bc991bc104d3a00952ce4bc166a4a1a"),
+		},
+		{
+			name:       "test 2",
+			storageKey: "0x0000000000000000000000000000000000000000000000000000000000000005",
+			want:       new(felt.Felt).SetHex("0x0000000000000000000000000000000000000000000000000000000000000005"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := tt.storageKey.Felt()
+			if !got.Equal(tt.want) {
+				t.Fatalf("StorageKey.Felt() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
```

Comment:
It's not clear what is being tested here. Serialisation of a storage key to a `felt.Felt`? The test names could also be more descriptive.

Reply:
Yes, tests the serialization of a storage key to a `felt.Felt`. I'll change the name to something more descriptive.

Reply:
Done here https://github.com/NethermindEth/juno/commit/3dfc627866e7bbace0bb0b2c06f049ecc011e30f